indexer-app/src/cmr/indexer/data/concepts/service.clj

```diff
@@ -0,0 +1,33 @@
+(ns cmr.indexer.data.concepts.service
+  "Contains functions to parse and convert service and service association concepts."
+  (:require
+   [cmr.common.log :refer (debug info warn error)]
+   [cmr.indexer.data.concept-parser :as concept-parser]
+   [cmr.transmit.metadata-db :as mdb]))
+
+(defn- service-association->service-concept
+  "Returns the service concept and service association for the given service association."
+  [context service-association]
+  (let [{:keys [service-concept-id]} service-association
+        service-concept (mdb/find-latest-concept
+                         context
+                         {:concept-id service-concept-id
+                          :latest true}
+                         :service)]
+    (when-not (:deleted service-concept)
+      service-concept)))
+
+(defn- has-formats?
+  "Returns true if the given service has more than one SupportedFormats value."
+  [context service-concept]
+  (let [service (concept-parser/parse-concept context service-concept)
+        supported-formats (get-in service [:ServiceOptions :SupportedFormats])]
+    (> (count supported-formats) 1)))
+
+(defn service-associations->elastic-doc
+  "Converts the service association into the portion going in the collection elastic document."
+  [context service-associations]
+  (let [service-concepts (remove nil?
+                                 (map #(service-association->service-concept context %)
+                                      service-associations))]
+    {:has-formats (boolean (some #(has-formats? context %) service-concepts))}))
```

Comment:
I don't particularly understand this, but I think its probably fine.  Its just weird that the ```service-associations->elastic-doc``` just returns a map with the single key :has-formats

Reply:
The return value of this function is merged with the collection doc map and passed to elastich to be indexed in elasticsearch. Also in the future we will have more fields (e.g. has_transform) in the map.